compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -4844,7 +4849,9 @@ public BType getSafeType(BType type, boolean liftNil, boolean liftError) {
         }
 
         if (liftError) {
-            errorLiftedType.remove(symTable.errorType);
+            memTypes = errorLiftedType.getMemberTypes().stream().filter(t -> t.tag != TypeTags.ERROR)
+                    .collect(Collectors.toCollection(LinkedHashSet::new));
+            errorLiftedType = BUnionType.create(null, memTypes);
```

Comment:
We need to check the size of `memTypes` before creating a union type. We shouldn't be creating a union type when there are 0 or 1 members. 0 should probably be neverType or semanticError.

Reply:
@MaryamZi in the next few lines, we are checking if the union type has one type or not and then return only that. We might need to return neverType if 0. That logic is not there.

Reply:
Ah, missed that. But we can probably avoid creating the union type unnecessarily, right? We can do this separately though.